python/tidy/servo_tidy/tidy.py

```diff
@@ -723,13 +723,35 @@ def parse_config(content):
     config['ignore']['directories'] = map(lambda path: os.path.join(*path.split('/')),
                                           config['ignore']['directories'])
 
+    # Add dict of dir, list of expected ext to config
+    dirs_to_check = config_file.get("check_ext", {})
+    config['check_ext'] = {}
+    # Fix the paths (OS-dependent)
+    for path, exts in dirs_to_check.items():
+        fixed_path = os.path.join(*path.split('/'))
+        config['check_ext'][fixed_path] = exts
+
     # Override default configs
     user_configs = config_file.get("configs", [])
     for pref in user_configs:
         if pref in config:
             config[pref] = user_configs[pref]
 
 
+def check_directory_files(only_changed_files):
+    for directory, file_extensions in config['check_ext'].items():
+        files = get_file_list(directory, only_changed_files, config["ignore"]["directories"])
+        for filename in files:
+            if not any(filename.endswith(ext) for ext in file_extensions):
+                details = {
+                    "name": os.path.basename(filename),
+                    "ext": ", ".join(file_extensions),
+                    "dir_name": os.path.dirname(filename)
+                }
+                message = "Unexpected extension found for {name}. We only expect {ext} in {dir_name}".format(**details)
```

Comment:
Nit: Let's be a bit more clear: `"We only expect files with {ext} extensions in {dir_name}"`